### PR TITLE
do subject alt names automatically

### PR DESCRIPTION
**Pull request by joejulian**
**What type of PR is this?**

chore

**What this PR does/ why we need it**:

My prior change required manually setting the dnsNames. This more-reliably sets them automatically.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: stable/dex/templates/certificate.yaml [hunk 1]
```diff
@@ -35,4 +35,10 @@ spec:
   organization:
     - D2iQ
 {{- end }}
-  dnsNames: {{ toYaml .Values.dnsNames | nindent 4 }}
+  dnsNames:
+    # names are the <name of the service>.<namespace>.srv.<domain> and
```
Comment (gracedo) [inline stable/dex/templates/certificate.yaml:39]:
```suggestion
    # names are the <name of the service>.<namespace>.svc.<domain> and
```

File: stable/dex/templates/certificate.yaml [hunk 2]
```diff
@@ -35,4 +35,10 @@ spec:
...
+    # shortened variations. However, there is no way to determine the domain,
+    # so we only include the up to svc.
```
Comment (gracedo) [inline stable/dex/templates/certificate.yaml:41]:
```suggestion
    # so we only include up to svc.
```

File: staging/kube-oidc-proxy/templates/certificate.yaml [hunk 3]
```diff
@@ -27,4 +27,10 @@ spec:
   organization:
     - D2iQ
 {{- end }}
-  dnsNames: {{ toYaml .Values.dnsNames | nindent 4 }}
+  dnsNames:
+    # names are the <name of the service>.<namespace>.srv.<domain> and
```
Comment (gracedo) [inline staging/kube-oidc-proxy/templates/certificate.yaml:31]:
```suggestion
    # names are the <name of the service>.<namespace>.svc.<domain> and
```

File: staging/kube-oidc-proxy/templates/certificate.yaml [hunk 4]
```diff
@@ -27,4 +27,10 @@ spec:
...
+    # shortened variations. However, there is no way to determine the domain,
+    # so we only include the up to svc.
```
Comment (gracedo) [inline staging/kube-oidc-proxy/templates/certificate.yaml:33]:
```suggestion
    # so we only include up to svc.
```

**Review by gracedo — APPROVED** [review]

**Review by alejandroEsc — APPROVED** [review]
lgtm

**Comment by alejandroEsc** [comment]
@joejulian are we not doing this? if so feel free to close though I suspect it may still be a good idea?

**Pull request merged**